src/components/movie-detail.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from "react";
+import '../movie.css';
+type Props = {
```

Comment:
Type the props like this
`type Props = RouteComponentProps<{id: string}>;`

`RouteComponentProps` can be imported from `react-router`
`id` is the name of the param in the url you choose

Basically now you can use `this.props.match` and other props that are provided by react router